ProcessQueue/__init__.py

```diff
@@ -111,161 +155,189 @@ def main(msg: func.QueueMessage):
+                "is_patient_id": False,
+                "is_date_event": False,
+                "is_ignore": False,
+                "pass_from_source": True,
+                "classification_system": str(ws.cell(row=i, column=12).value),
                 "date_type": "",
                 "concept_id": -1,
                 "field_description": None,
             }
-            # Append each entry to list
+            # Append each entry to a list
             data.append(scan_report_field_entry)
-            # If there was an empty row remove it from list,
-            # Switch to writing to the next table
+            # If there is an empty row(end of a table) POST fields in this table
             if not any(cell.value for cell in row_cell):
-                idx=idx+1
+                # .pop() empty row from list,
                 data.pop()
-        # Create JSON array
-        json_data=json.dumps(data)
-        # POST request
-        response = requests.post("{}scanreportfields/".format(api_url), data=json_data,headers=headers)
-        print('FIELD SAVE STATUS >>>', response.status_code)
-        # Load result from the response,
-        # Save generated field ids, and the corresponding name
-        response=json.loads(response.content.decode("utf-8"))
-        
-        for element in range(len(response)):
-            field_ids.append(str(response[element].get('id',None)))
-            field_names.append(str(response[element].get('name',None)))
-        
-        # Fix for empty row (empty row was being returned as [] element)
-        # Need to remove for dict(zip()) to work
-        for name in field_names:
-            if name.startswith("[") or None:
-              field_names.remove(name)
-        for id in field_ids:
-            if id=="None": field_ids.remove(id)
-        # Create a dictionary with field names and field ids 
-        # as key value pairs
-        # e.g ("Field Name":<Field ID>)
-        namexids=dict(zip(field_names,field_ids))
+                # JSON Array for fields in current table
+                json_data = json.dumps(data)
+                # POST Fields
+                response = requests.post(
+                    "{}scanreportfields/".format(api_url),
+                    data=json_data,
+                    headers=headers,
+                )
+                print("FIELD SAVE STATUS >>>", response.status_code)
+                # Load result from the response,
+                # Save generated field ids, and the corresponding name
+                response = json.loads(response.content.decode("utf-8"))
+                for element in range(len(response)):
+                    field_ids.append(str(response[element].get("id", None)))
+                    field_names.append(str(response[element].get("name", None)))
+                # Fix for empty row (empty row was being returned as [] element)
+                # Need to remove for dict(zip()) to work
+                for name in field_names:

```

Comment:
In this `for` loop and the next, can we be _absolutely_ certain that all `"["` or `None` in `field_ids` will match up exactly with all `"None"` in `field_ids`? If not, we could end up misaligning them. Is there a safer way to do this? (perhaps iterate over both at the same time with a `zip` and only delete when both match the criteria - I just think this would be safer and avoid some difficult-to-notice bug with misaligned lists).

Reply:
Yeah sure I can try that, I don't remember what exactly was causing this or why I fixed it like this so I'll have a look and see if this is still an issue with the changed code in the first place

Reply:
Sure - if we don't get those same "[" or "None" outputs then we can get rid - otherwise, let's go with the safer implementation.